src/line.cpp

```diff
@@ -526,92 +526,108 @@ std::vector<point> squares_in_direction( const int x1, const int y1, const int x
     return adjacent_squares;
 }
 
-float rl_vec2d::norm()
+float rl_vec2d::magnitude() const
 {
     return sqrt(x * x + y * y);
 }
 
-float rl_vec3d::norm()
+float rl_vec3d::magnitude() const
 {
     return sqrt(x * x + y * y + z * z);
 }
 
-rl_vec2d rl_vec2d::normalized()
+rl_vec2d rl_vec2d::normalized() const
 {
     rl_vec2d ret;
     if (is_null()) { // shouldn't happen?
         ret.x = ret.y = 1;
         return ret;
     }
-    float n = norm();
-    ret.x = x / n;
-    ret.y = y / n;
+    const float m = magnitude();
+    ret.x = x / m;
+    ret.y = y / m;
     return ret;
 }
 
-rl_vec3d rl_vec3d::normalized()
+rl_vec3d rl_vec3d::normalized() const
 {
     rl_vec3d ret;
     if (is_null()) { // shouldn't happen?
-        ret.x = ret.y = ret.z = 1;
+        ret.x = ret.y = ret.z = 0;
         return ret;
     }
-    float n = norm();
-    ret.x = x / n;
-    ret.y = y / n;
-    ret.z = z / n;
+    const float m = magnitude();
+    ret.x = x / m;
+    ret.y = y / m;
+    ret.z = z / m;
     return ret;
 }
 
-rl_vec2d rl_vec2d::get_vertical()
+rl_vec2d rl_vec2d::rotated( float angle ) const
 {
-    rl_vec2d ret;
-    ret.x = -y;
-    ret.y = x;
-    return ret;
+    return rl_vec2d(
+        x * cos( angle ) - y * sin( angle ),
+        x * sin( angle ) + y * cos( angle )
+    );
 }
 
-rl_vec3d rl_vec3d::get_vertical()
+rl_vec3d rl_vec3d::rotated( float angle ) const
 {
-    rl_vec3d ret;
-    ret.x = -y;
-    ret.y = x;
-    ret.z = z;
-    return ret;
+    return rl_vec3d(
+        x * cos( angle ) - y * sin( angle ),
+        x * sin( angle ) + y * cos( angle )
+    );
 }
 
-float rl_vec2d::dot_product (rl_vec2d &v)
+float rl_vec2d::dot_product (rl_vec2d &v) const
 {
     float dot = x * v.x + y * v.y;
     return dot;
 }
 
-float rl_vec3d::dot_product (rl_vec3d &v)
+float rl_vec3d::dot_product (rl_vec3d &v) const
```

Comment:
As you're changing the signature of this function anyway: it should take a const reference.